drivers/counter/counter_nrfx_timer.c

```diff
@@ -174,13 +173,24 @@ static int counter_nrfx_set_top_value(struct device *dev, u32_t ticks,
 	}
 
 	nrfx_timer_compare_int_disable(timer, TOP_CH);
-	nrfx_timer_clear(timer);
+	if (!(cfg->flags & COUNTER_TOP_CFG_DONT_RESET)) {
+		nrfx_timer_clear(timer);
+	}
 
-	data->top_cb = callback;
-	data->top_user_data = user_data;
+	data->top_cb = cfg->callback;
+	data->top_user_data = cfg->user_data;
 	nrfx_timer_extended_compare(timer, TOP_CH,
-				    ticks, COUNTER_OVERFLOW_SHORT,
-				    callback ? true : false);
+				    cfg->ticks, COUNTER_OVERFLOW_SHORT,
+				    cfg->callback ? true : false);
+
+	if ((cfg->flags & COUNTER_TOP_CFG_DONT_RESET) &&
```

Comment:
Actually, even if the counter is reset, we can end up with the top value set lower than the current counter value. Although it could happen only for very low top values and accordingly high latencies between clearing the counter and setting the new compare value, but maybe such unusual scenario should be handled as well?

Reply:
fixed

Reply:
Cool. Should we use the same approach in `counter_nrfx_rtc`, for consistency and for people having the need to set top value to `1`?

Reply:
done